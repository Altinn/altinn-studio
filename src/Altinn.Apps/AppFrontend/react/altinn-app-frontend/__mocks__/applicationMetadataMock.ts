const applicationMetadataJSON = `{
  "id": "mockOrg/test-app",
  "org": "mockOrg",
  "title": {
    "nb": "Test App"
  },
  "dataTypes": [
    {
      "id": "test-data-model",
      "allowedContentTypes": [
        "application/xml"
      ],
      "appLogic": {
        "autoCreate": true,
        "classRef": "Altinn.App.Models.Skjema"
      },
      "taskId": "Task_1",
      "maxCount": 1,
      "minCount": 1
    },
    {
      "id": "ref-data-as-pdf",
      "allowedContentTypes": [
        "application/pdf"
      ],
      "maxCount": 0,
      "minCount": 0
    },
    {
      "id": "test-data-type-1",
      "allowedContentTypes": [
        "application/pdf"
      ],
      "maxCount": 5,
      "minCount": 0
    },
    {
      "id": "test-data-type-2",
      "allowedContentTypes": [
        "application/pdf"
      ],
      "maxCount": 2,
      "minCount": 0
    }
  ],
  "partyTypesAllowed": {
    "bankruptcyEstate": false,
    "organisation": false,
    "person": true,
    "subUnit": false
  },
  "created": "2020-06-29T08:47:12.425551Z",
  "createdBy": "test testesen",
  "lastChanged": "2020-06-29T08:47:12.4255537Z",
  "lastChangedBy": "test testesen"
}`;

export const applicationMetadataMock = JSON.parse(applicationMetadataJSON);
