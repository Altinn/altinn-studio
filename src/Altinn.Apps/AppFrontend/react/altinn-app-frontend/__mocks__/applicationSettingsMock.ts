const applicationSettingsJSON = `{
  "id": "mockOrg/test-app",
  "org": "mockOrg",
  "title": {
    "nb": "Test App"
  }}`;

export const applicationSettingsMock = JSON.parse(applicationSettingsJSON);
