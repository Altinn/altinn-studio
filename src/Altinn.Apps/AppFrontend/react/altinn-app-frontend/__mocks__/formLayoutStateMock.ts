import { ILayoutState } from '../src/features/form/layout/formLayoutReducer';

export function getFormLayoutStateMock(customStates?: Partial<ILayoutState>): ILayoutState {
  const mockFormLayoutState: ILayoutState = {
    layouts: {
      FormLayout: [
        {
          id: 'field1',
          type: 'Input',
          dataModelBindings: {
            simple: 'Group.prop1',
          },
          textResourceBindings: {
            title: 'Title',
          },
          readOnly: false,
          required: false,
          disabled: false,
        },
        {
          id: 'field2',
          type: 'Input',
          dataModelBindings: {
            simple: 'Group.prop2',
          },
          textResourceBindings: {
            title: 'Title',
          },
          readOnly: false,
          required: false,
          disabled: false,
        },
        {
          id: 'field3',
          type: 'Input',
          dataModelBindings: {
            simple: 'Group.prop3',
          },
          textResourceBindings: {
            title: 'Title',
          },
          readOnly: false,
          required: false,
          disabled: false,
        },
      ],
    },
    error: {
      message: null,
      name: null,
    },
    uiConfig: {
      autoSave: true,
      focus: null,
      hiddenFields: [],
      repeatingGroups: null,
      currentView: 'FormLayout',
      navigationConfig: {},
      layoutOrder: [],
    },
  };

  return {
    ...mockFormLayoutState,
    ...customStates,
  };
}
