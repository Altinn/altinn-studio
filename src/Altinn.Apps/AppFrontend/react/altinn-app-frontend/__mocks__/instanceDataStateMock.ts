import { IInstanceDataState } from '../src/shared/resources/instanceData/instanceDataReducers';

export function getInstanceDataStateMock(customStates?: Partial<IInstanceDataState>): IInstanceDataState {
  const mockInstanceDataState = {
    error: null,
    instance: {
      instanceOwner: {
        partyId: '12345',
        personNumber: '01017512345',
        organisationNumber: null,
      },
      appId: 'mockOrg/mockApp',
      created: new Date('2020-01-01'),
      data: [],
      id: '91cefc5e-c47b-40ff-a8a4-05971205f783',
      instanceState: null,
      lastChanged: new Date('2020-01-01'),
      org: 'mockOrg',
      process: {
        started: '2020-01-01',
        startEvent: null,
        currentTask: {
          flow: 1,
          started: '2020-01-01',
          elementId: 'mockElementId',
          name: 'Task_1',
          altinnTaskType: 'FormFilling',
          ended: null,
          validated: null,
        },
        ended: null,
        endEvent: null,
      },
      selfLinks: null,
      status: null,
      title: null,
    },
  };

  return {
    ...mockInstanceDataState,
    ...customStates,
  };
}
