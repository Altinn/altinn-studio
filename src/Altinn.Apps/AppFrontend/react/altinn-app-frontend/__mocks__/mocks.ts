export { getFormLayoutStateMock } from './formLayoutStateMock';
export { getInitialStateMock } from './initialStateMock';
export { getFormDataStateMock } from './formDataStateMock';
export { getLanguageFromCode as getLanguageMock } from '../../shared/src/language';
export { applicationMetadataMock } from './applicationMetadataMock';
