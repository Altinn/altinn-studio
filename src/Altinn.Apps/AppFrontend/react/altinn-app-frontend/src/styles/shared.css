.validation-error {
  border: 2px solid #E23B53 !important;
}
.disabled-date {
  border: none !important;
  background-color: transparent !important;
  color: #000 !important;
}
.disabled {
  border: 2px solid #6A6A6A !important;
  color: #000;
  background: repeating-linear-gradient(
    135deg,
    #efefef,
    #efefef 2px,
    #fff 3px,
    #fff 5px
  );
}
input:hover:not(.disabled) {
  border-color: #0062ba !important;
}

input.form-control:focus:not(.a-hasButton) {
  outline: 2px solid #0062ba;
  outline-offset: 0px;
}
.input-group:focus-within {
  outline: 2px solid #0062ba;
  outline-offset: 0px;
}
