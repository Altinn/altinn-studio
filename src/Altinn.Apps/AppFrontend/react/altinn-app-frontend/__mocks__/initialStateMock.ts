import { getLanguageFromCode } from '../../shared/src/language';
import { IRuntimeState } from '../src/types';
import { getFormLayoutStateMock } from './formLayoutStateMock';
import { getFormDataStateMock } from './formDataStateMock';
import { applicationMetadataMock } from './applicationMetadataMock';
import { IParty } from '../../shared/src';
import { getInstanceDataStateMock } from './instanceDataStateMock';
import { getProfileStateMock } from './profileStateMock';

export const mockParty: IParty = {
  partyId: '12345',
  name: 'Ola Privatperson',
  ssn: '01017512345',
  partyTypeName: null,
  orgNumber: null,
  unitType: null,
  isDeleted: false,
  onlyHierarchyElementWithNoAccess: false,
  childParties: null,
};

export function getInitialStateMock(customStates?: Partial<IRuntimeState>): IRuntimeState {
  const initialState: IRuntimeState = {
    applicationMetadata: {
      applicationMetadata: applicationMetadataMock,
      error: null,
    },
    attachments: {
      attachments: null,
    },
    formData: getFormDataStateMock(),
    formDataModel: {
      dataModel: [],
      error: null,
      schemas: null,
    },
    formDynamics: {
      apis: null,
      conditionalRendering: null,
      error: null,
      ruleConnection: null,
    },
    formLayout: getFormLayoutStateMock(),
    formRules: {
      error: null,
      fetched: false,
      fetching: false,
      model: null,
    },
    formValidations: {
      validations: {},
      error: null,
      invalidDataTypes: [],
      currentSingleFieldValidation: null,
    },
    instanceData: getInstanceDataStateMock(),
    instantiation: {
      error: null,
      instanceId: null,
      instantiating: null,
    },
    isLoading: {
      dataTask: false,
    },
    language: {
      language: getLanguageFromCode('nb'),
      error: null,
    },
    organisationMetaData: {
      allOrgs: null,
      error: null,
    },
    party: {
      error: null,
      parties: [
        mockParty,
      ],
      selectedParty: mockParty,
    },
    process: {
      error: null,
      state: null,
    },
    profile: getProfileStateMock(),
    queue: {
      appTask: null,
      dataTask: null,
    },
    textResources: {
      resources: [],
      error: null,
      language: 'nb',
    },
    optionState: {
      options: {},
      error: null,
    },
  };

  return {
    ...initialState,
    ...customStates,
  };
}
