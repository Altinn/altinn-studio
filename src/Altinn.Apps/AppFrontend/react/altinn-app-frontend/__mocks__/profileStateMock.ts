import { IProfileState } from '../src/shared/resources/profile/profileReducers';
import { mockParty } from './initialStateMock';

export function getProfileStateMock(customStates?: Partial<IProfileState>): IProfileState {
  const profileStateMock = {
    error: null,
    profile: {
      userId: 12345,
      userName: 'Ola Normann',
      partyId: 12345,
      party: mockParty,
      userType: 1,
      profileSettingPreference: {
        language: 'nb',
        preSelectedPartyId: 12345,
        doNotPromptForParty: false,
      },
    },
  };

  return {
    ...profileStateMock,
    ...customStates,
  };
}
