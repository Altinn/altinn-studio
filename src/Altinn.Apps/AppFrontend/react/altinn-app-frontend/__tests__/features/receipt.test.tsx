// tslint:disable: max-line-length
import 'jest';
import * as React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { getInitialStateMock } from '../../__mocks__/mocks';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import ReceiptContainer, { returnInstanceMetaDataObject } from '../../src/features/receipt/containers/receiptContainer';
import { IData } from '../../../shared/src';

function mockDataElements() : IData[] {
  return [
    {
      id: '1C75319C-DBA2-485D-9EF7-BE3D0F4F0B5B',
      contentType: 'application/pdf',
      dataType: 'ref-data-as-pdf',
      filename: 'data-as.pdf',
      locked: true,
      refs: [],
      selfLinks: {
        apps: 'applink',
        platform: "platformlink",
      },
      blobStoragePath: 'blobpath',
      size: 23,
      created: new Date('2020-01-01'),
      createdBy: 'me',
      lastChanged: new Date('2020-01-01'),
      lastChangedBy: 'me'
    }
  ]
}

describe('Testing functions in ReceiptContainer', () => {
  let mockStore: any;

  test('returnInstanceMetaDataObject() returns correct object', () => {
    const testData = {
      orgsData: {
        tdd: {
          name: {
            en: 'Test Ministry',
            nb: 'Testdepartementet',
            nn: 'Testdepartementet',
          },
          logo: '',
          orgnr: '',
          homepage: '',
        },
        ttd: {
          name: {
            en: 'Test Ministry',
            nb: 'Testdepartementet',
            nn: 'Testdepartementet',
          },
          logo: '',
          orgnr: '',
          homepage: '',
        },
      },
      languageData: null,
      profileData: {
        profile: {
          userId: 1,
          userName: 'OlaNordmann',
          phoneNumber: '90012345',
          email: 'ola@altinncore.no',
          partyId: 50001,
          party: {
            partyId: 50001,
            partyTypeName: 1,
            orgNumber: null,
            ssn: null,
            unitType: null,
            name: 'Ola Privatperson',
            isDeleted: false,
            onlyHierarchyElementWithNoAccess: false,
            person: {
              ssn: '01017512345',
              name: null,
              firstName: 'Ola',
              middleName: null,
              lastName: 'Privatperson',
              telephoneNumber: null,
              mobileNumber: null,
              mailingAddress: null,
              mailingPostalCode: null,
              mailingPostalCity: null,
              addressMunicipalNumber: null,
              addressMunicipalName: null,
              addressStreetName: null,
              addressHouseNumber: null,
              addressHouseLetter: null,
              addressPostalCode: null,
              addressCity: null,
            },
            organisation: null,
          },
          userType: 1,
          profileSettingPreference: null,
        },
      },
      instanceGuid: '6697de17-18c7-4fb9-a428-d6a414a797ae',
      userLanguageString: 'nb',
      lastChangedDateTime: '22.08.2019 / 09:08',
      instance: {
        org: 'tdd',
      },
      instanceOwnerParty: {
        partyId: 50001,
        name: 'Ola Privatperson',
        ssn: '01017512345'
      }
    };

    const expected = {
      'receipt.date_sent': '22.08.2019 / 09:08',
      'receipt.receiver': 'Testdepartementet',
      'receipt.ref_num': 'd6a414a797ae',
      'receipt.sender': '01017512345-Ola Privatperson',
    };

    expect(returnInstanceMetaDataObject(
      testData.orgsData,
      testData.languageData,
      testData.instanceOwnerParty,
      testData.instanceGuid,
      testData.userLanguageString,
      testData.lastChangedDateTime,
      testData.instance.org,
    )).toEqual(expected);
  });

  it('+++ KAJSDLAS ASD ASD ASD ASD ASD ASD ASD ', () => {
    const createStore = configureStore();

    const newState = getInitialStateMock();
    newState.applicationMetadata.applicationMetadata.autoDeleteOnProcessEnd = true;
    newState.instanceData.instance.data = mockDataElements();

    mockStore = createStore(newState);
    const wrapper = mount(
      <MemoryRouter>
        <Provider store={mockStore}>
          <ReceiptContainer />
        </Provider>
      </MemoryRouter>,
    );
    expect(wrapper).toMatchSnapshot();
  });

});
