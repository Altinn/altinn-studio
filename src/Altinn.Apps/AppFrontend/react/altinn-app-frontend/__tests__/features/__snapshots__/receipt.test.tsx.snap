// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing functions in ReceiptContainer +++ KAJSDLAS ASD ASD ASD ASD ASD ASD ASD  1`] = `
<svg
  aria-labelledby="olqvtx-aria"
  height={561}
  id={null}
  role="img"
  style={Object {}}
  width={705}
>
  <title
    id="olqvtx-aria"
  >
    Loading...
  </title>
  <rect
    clipPath="url(#olqvtx-diff)"
    height="100%"
    role="presentation"
    style={
      Object {
        "fill": "url(#olqvtx-animated-diff)",
      }
    }
    width="100%"
    x="0"
    y="0"
  />
  <defs
    role="presentation"
  >
    <clipPath
      id="olqvtx-diff"
    >
      <rect
        height="32"
        rx="0"
        ry="0"
        width="442"
        x="12"
        y="11"
      />
      <rect
        height="0"
        rx="0"
        ry="0"
        width="16"
        x="322"
        y="37"
      />
      <rect
        height="10"
        rx="0"
        ry="0"
        width="6"
        x="65"
        y="77"
      />
      <rect
        height="0"
        rx="0"
        ry="0"
        width="0"
        x="304"
        y="85"
      />
      <rect
        height="98"
        rx="0"
        ry="0"
        width="444"
        x="12"
        y="74"
      />
      <rect
        height="26"
        rx="0"
        ry="0"
        width="448"
        x="12"
        y="191"
      />
      <rect
        height="76"
        rx="0"
        ry="0"
        width="610"
        x="12"
        y="245"
      />
      <rect
        height="30"
        rx="0"
        ry="0"
        width="191"
        x="12"
        y="346"
      />
      <rect
        height="31"
        rx="0"
        ry="0"
        width="101"
        x="12"
        y="413"
      />
      <circle
        cx="36"
        cy="474"
        r="25"
      />
      <rect
        height="27"
        rx="0"
        ry="0"
        width="181"
        x="72"
        y="460"
      />
    </clipPath>
    <linearGradient
      id="olqvtx-animated-diff"
    >
      <stop
        offset="0%"
        stopColor="#f5f6f7"
        stopOpacity={1}
      >
        <animate
          attributeName="offset"
          dur="1.2s"
          keyTimes="0; 0.25; 1"
          repeatCount="indefinite"
          values="-2; -2; 1"
        />
      </stop>
      <stop
        offset="50%"
        stopColor="#eee"
        stopOpacity={1}
      >
        <animate
          attributeName="offset"
          dur="1.2s"
          keyTimes="0; 0.25; 1"
          repeatCount="indefinite"
          values="-1; -1; 2"
        />
      </stop>
      <stop
        offset="100%"
        stopColor="#f5f6f7"
        stopOpacity={1}
      >
        <animate
          attributeName="offset"
          dur="1.2s"
          keyTimes="0; 0.25; 1"
          repeatCount="indefinite"
          values="0; 0; 3"
        />
      </stop>
    </linearGradient>
  </defs>
</svg>
`;
